### DISCUSSION TITLE
Cask `element` fake update process

### DISCUSSION THREAD

**Discussion by qupig** [Homebrew/discussions#5393] [Casks] [answered]
### Description of issue

```console
% brew upgrade element
==> Upgrading 1 outdated package:
element 1.11.66 -> 1.11.67
==> Upgrading element
==> Downloading https://packages.element.io/desktop/install/macos/Element.dmg
Already downloaded: /Users/qupig/Library/Caches/Homebrew/downloads/0a10f939450ebf7bf3ef038b28bc84002cbd05f784ef8e8d24246bfe366bf18f--Element.dmg
Warning: No checksum defined for cask 'element', skipping verification.
==> Backing App 'Element.app' up to '/opt/homebrew/Caskroom/element/1.11.66/Element.app'
==> Removing App '/Applications/Element.app'
==> Moving App 'Element.app' to '/Applications/Element.app'
==> Purging files for version 1.11.66 of Cask element
🍺  element was successfully upgraded!
```

As you can see, the cask installer was not re-downloaded due to the absence of checksum, and the same URL.

As a result, the version of element is still 1.11.66 after prompting that the update is successful.

It is recommended to optimize the check verification mechanism or update process.

You can see that the official actually provides two formats of URLs, with or without version numbers:

https://packages.element.io/desktop/install/macos/index.html
https://packages.element.io/desktop/install/macos/Element.dmg
https://packages.element.io/desktop/install/macos/Element-1.11.66-universal.dmg
https://packages.element.io/desktop/install/macos/Element-1.11.67-universal.dmg

### Relevant casks

homebrew/cask/element
https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/e/element.rb

**Comment by SMillerDev** [comment]
Please create a pull request to replace the url with the versioned URL.

**Comment by krehel** [comment, marked as answer]
https://github.com/Homebrew/homebrew-cask/pull/174625
